components/Common/Select/index.tsx

```diff
@@ -20,12 +29,33 @@ const Select: FC<SelectProps> = ({
   placeholder,
   label,
   dropdownLabel,
+  inline,
   onChange,
 }) => {
   const id = useId();
+  const mappedValues: SelectValue[] = useMemo(() => {
+    if (!values.length) {
+      return [];
+    }
+
+    const [firstItem, ...items] = values;
+
+    if (typeof firstItem === 'string') {
+      return [firstItem, ...(items as string[])].map(value => ({
```

Comment:
I don't think you need to do type casting here, why is the type casting needed to begin. with? TypeScript should already be able to infer that values is either string or something.

Could you share the errors?

Reply:
#### When removing `as string[]`;

```
Type '{ value: string | SelectValue; label: string | SelectValue; }[]' is not assignable to type 'SelectValue[]'.
  Type '{ value: string | SelectValue; label: string | SelectValue; }' is not assignable to type 'SelectValue'.
    Types of property 'label' are incompatible.
      Type 'string | SelectValue' is not assignable to type 'string'.
        Type 'SelectValue' is not assignable to type 'string'.ts(2322)
```

#### When removing `as SelectValue[]`;

```
Type '(string | SelectValue)[] | { value: string; label: string; }[]' is not assignable to type 'SelectValue[]'.
  Type '(string | SelectValue)[]' is not assignable to type 'SelectValue[]'.
    Type 'string | SelectValue' is not assignable to type 'SelectValue'.
      Type 'string' is not assignable to type 'SelectValue'.ts(2322)
```

When we deconstruct an array with a single type, it really looks interesting to specify the type for the part where we do not type-check

Reply:
Right I guess we can do:

```tsx
const mappedItems = useMemo(() => {
  const [firstItem] = items;

  const _mappedItems = 
    typeof firstItem === 'string' ?
      items.map(value => ({ value, label: value }) :
      items;

  return _mappedItems as SelectValue[];
}, [items])
```